core/deployment/src/test/java/io/quarkiverse/langchain4j/deployment/GuardrailExecutionProcessorTest.java

```diff
@@ -0,0 +1,344 @@
+package io.quarkiverse.langchain4j.deployment;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.stream.Stream;
+
+import jakarta.enterprise.context.ApplicationScoped;
+
+import org.jboss.jandex.ClassInfo;
+import org.jboss.jandex.Index;
+import org.jboss.jandex.MethodInfo;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.junit.jupiter.params.provider.ValueSource;
+
+import dev.langchain4j.data.message.AiMessage;
+import dev.langchain4j.data.message.UserMessage;
+import io.micrometer.core.annotation.Counted;
+import io.micrometer.core.annotation.Timed;
+import io.opentelemetry.instrumentation.annotations.WithSpan;
+import io.quarkiverse.langchain4j.RegisterAiService;
+import io.quarkiverse.langchain4j.deployment.GuardrailExecutionProcessor.TransformType;
+import io.quarkiverse.langchain4j.guardrails.Guardrail;
+import io.quarkiverse.langchain4j.guardrails.GuardrailParams;
+import io.quarkiverse.langchain4j.guardrails.GuardrailResult;
+import io.quarkiverse.langchain4j.guardrails.InputGuardrail;
+import io.quarkiverse.langchain4j.guardrails.InputGuardrailParams;
+import io.quarkiverse.langchain4j.guardrails.InputGuardrailResult;
+import io.quarkiverse.langchain4j.guardrails.InputGuardrails;
+import io.quarkiverse.langchain4j.guardrails.OutputGuardrail;
+import io.quarkiverse.langchain4j.guardrails.OutputGuardrailParams;
+import io.quarkiverse.langchain4j.guardrails.OutputGuardrailResult;
+import io.quarkiverse.langchain4j.guardrails.OutputGuardrails;
+
+class GuardrailExecutionProcessorTest {
```

Comment:
We usually don't do these kinds of tests - they really hinder refactoring. We generally use `@QuarkusUnitTest` with a large set of representative "applications"

Reply:
I get it (& I did that as well), but since this was pretty complicated class walking I wanted to be able to unit test lots of different edge cases.

Reply:
Just a note - the tests I wrote don't require the `BuildStep`s to run. They're just testing some of the helper methods in the processor class.

Reply:
> I could refactor those methods out into a helper class if you'd like and then the tests could be associated to the helper class.

That would be nice

Reply:
Done!